### ISSUE TITLE
Pair struct isn't understood properly by generator (has a union member)

### ISSUE THREAD

**Issue by timo** [cimgui/cimgui#88]
https://github.com/cimgui/cimgui/blob/master/generator/output/structs_and_enums.json#L2750

```
    "Pair": [
      {
        "name": "key",
        "type": "ImGuiID"
      },
      {
        "name": "}",
        "type": "union { int val_i; float val_f; void* val_p;"
      }
    ],
```

Clearly both the name and type of the second member isn't quite right.

**Comment by sonoro1234** [comment]
In first place this indefinition is generated in imgui.h as
https://github.com/ocornut/imgui/blob/7a5058e3bfea4149acef2ad98d8fdd62605a0f4c/imgui.h#L1604

Is as unamed member of Pair struct.
This is legal C and Cpp so that
```cpp
Pair foo;
foo.val_i = 32;
```
is valid code.
But for our parsing is a problem

This should be 
```cpp
union { int val_i; float val_f; void* val_p; } type;
```
in imgui.h and then
```
    "Pair": [
      {
        "name": "key",
        "type": "ImGuiID"
      },
      {
        "name": "type",
        "type": "union { int val_i; float val_f; void* val_p;}"
      }
    ],
```
would be generated.

but then in Cpp it shoud be used as
```cpp
Pair foo;
foo.type.val_i = 32;
```

Solution could be perhaps to define Pair as

```cpp
    struct Pair
    {
        ImGuiID key;
        union { int val_i; float val_f; void* val_p; } type;
        Pair(ImGuiID _key, int _val_i)   { key = _key; type.val_i = _val_i; }
        Pair(ImGuiID _key, float _val_f) { key = _key; type.val_f = _val_f; }
        Pair(ImGuiID _key, void* _val_p) { key = _key; type.val_p = _val_p; }
    };
```

but I am not completely sure about it.
@ocornut Should review that

**Comment by ocornut** [comment]
Couldn't the parsing be fixed so that the output eventually work?
I'm using the current syntax in variety of code path and anyone using `ImGuiStorager` as a simple-portable-dumb `map<>` replacement would have their code broken if we change it.

**Comment by sonoro1234** [comment]
The output of cimgui.cpp 
https://github.com/cimgui/cimgui/blob/master/cimgui.cpp#L1610
is ok

The output of
https://github.com/cimgui/cimgui/blob/master/generator/output/structs_and_enums.json#L2750
cant be fixed because we just dont have any name to output for the type.

**Comment by ocornut** [comment]
Well you could just output:

```
{
        "name": "",
        "type": "union { int val_i; float val_f; void* val_p;"
}
```

Even with a name, outputting a "union ..." string is going to be a very specific thing that will not map to other languages either way. This could perfectly be private/ignored as bindings don't need to expose it.

**The question is more what does @timo expect to see there that will be useful in their context?**
My intuition is that NOT outputting the block will be better.
Otherwise make it a requirement that the code using that data knows what a "union ..." string means.

**Comment by timo** [comment]
I'd find an empty name less surprising than having a piece of syntax leak from the type field into the name field, that much i'm sure about :)

I don't think I've seen any other instance of anonymous unions in the json output files yet, so I might as well just special-case it.

**Comment by ocornut** [comment]
If the converter can detect it, perhaps it could output a different field e.g.
```
{
  "raw_c" : "union { int val_i; float val_f; void* val_p;"
}
```
Which cimgui would let thru but other parsers/converters would ignore?

**Comment by sonoro1234** [comment]
Now corrected in cimgui ImVector branch (will be merged)

**Comment by ocornut** [comment]
Excellent. I'll release 1.68 today or tomorrow so it's good that cimgui is ready for it.

**Comment by sonoro1234** [comment]
already merged in master 1.68
Still pending the question about unnamed struct fields being std=c11 but not std=c99
